node/src/components/small_network.rs

```diff
@@ -152,22 +157,30 @@ where
     /// Tracks whether a connection is symmetric or not.
     connection_symmetries: HashMap<NodeId, ConnectionSymmetry>,
 
-    /// Tracks nodes that have announced themselves as joining nodes.
-    joining_nodes: HashSet<NodeId>,
+    /// Tracks nodes that have announced themselves as nodes that are syncing.
+    syncing_nodes: HashSet<NodeId>,
 
-    /// Channel signaling a shutdown of the small network.
+    /// Channel signaling a shutdown of the server.
     // Note: This channel is closed when `SmallNetwork` is dropped, signalling the receivers that
-    // they should cease operation.
+    // they should cease operation. It is also replaced with a new channel when we finished the
```

Comment:
This is not correct - we should _not_ be triggering the shutdown sender in any case anymore except when actually shutting down.